livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -162,21 +165,8 @@ public InterpreterResult interpret(String st, InterpreterContext context) {
 
   @Override
   public void cancel(InterpreterContext context) {
-    if (livyVersion.isCancelSupported()) {
-      String paraId = context.getParagraphId();
-      Integer stmtId = paragraphId2StmtIdMap.get(paraId);
-      try {
-        if (stmtId != null) {
-          cancelStatement(stmtId);
-        }
-      } catch (LivyException e) {
-        LOGGER.error("Fail to cancel statement " + stmtId + " for paragraph " + paraId, e);
-      } finally {
-        paragraphId2StmtIdMap.remove(paraId);
-      }
-    } else {
-      LOGGER.warn("cancel is not supported for this version of livy: " + livyVersion);
-    }
+    paragraphsToCancel.add(context.getParagraphId());
```

Comment:
Since livyVersion is available only after the session is initialized, putting it inside the if block will work based on whether livyVersion is available or not. In some cases, this , might even throw NullPointerException.  

I think , it's better to simply add it and then remove it later. 
If the livyVersion supports cancellation, we will cancel and remove.
If not supported, we will warn and remove.